optimum/gptq/quantizer.py

```diff
@@ -0,0 +1,629 @@
+                for name in subset_layers:
+                    gptq[name] = GPTQ(subset_layers[name])
+                    gptq[name].quantizer.configure(bits=self.bits, sym=self.sym, perchannel=True)
+
+                    def add_batch(name):
+                        def tmp(_, input, output):
+                            gptq[name].add_batch(input[0].data, output.data)
+
+                        return tmp
+
+                    # because it adding a hook will replace the old one.
+                    handles.append(subset_layers[name].register_forward_hook(add_batch(name)))
+                # update Hessian for each layer in subset_layers thanks to the hook
+                for j in range(len(dataset)):
+                    # the args are already on the gpu
+                    # don't need to store the output
+                    block(layer_inputs[j], **layer_input_kwargs[j])
+                # remove hook
+                for h in handles:
+                    h.remove()
+                for name in subset_name_list:
+                    logger.info(f"Quantizing {name} in block {i + 1}/{len(blocks)}...")
+                    scale, zero, g_idx = gptq[name].fasterquant(
+                        percdamp=self.damp_percent, group_size=self.group_size, actorder=self.desc_act
+                    )
+                    # if we pack the model at the end
+                    if not self.pack_sequentially:
+                        quantizers[f"{self.block_name_to_quantize}.{i}.{name}"] = (
+                            gptq[name].quantizer,
+                            scale,
+                            zero,
+                            g_idx,
+                        )
+                    else:
+                        # put on cpu because it is not possible to quantize on cuda for now
+                        subset_layers[name], scale, zero, g_idx = (
+                            subset_layers[name].to("cpu"),
+                            scale.to("cpu"),
+                            zero.to("cpu"),
+                            g_idx.to("cpu"),
+                        )
+                        layer_name = f"{self.block_name_to_quantize}.{i}.{name}"
+                        self._replace_by_quant_layers(model, [layer_name])
+                        quantized_layer = recurse_getattr(model, layer_name)
+                        device_layer = get_device(quantized_layer)
+                        quantized_layer = quantized_layer.to("cpu")
+                        quantized_layer.pack(subset_layers[name], scale, zero, g_idx)
+                        quantized_layer = quantized_layer.to(device_layer)
+                    gptq[name].free()
+                del subset_layers
+            # we get the new output from the partial quantized block
+            for j in range(len(dataset)):
+                layer_output = block(layer_inputs[j], **layer_input_kwargs[j])[0]
+                layer_outputs.append(layer_output)
```

Comment:
This does not seem to be used

Reply:
it is used here : `layer_inputs, layer_outputs = layer_outputs, []`. See comment above.